### DISCUSSION TITLE
Bootstrap 5: Modal keeps open after History Back and cannot be closed

### DISCUSSION THREAD

**Discussion by chriskrams** [twbs/bootstrap#36810] [Q&A]
I have discovered a strange behavior with modal windows and need help there.

Bootstrap: 5.1.3
Ruby: 2.5.7
Rails: 6.1.4
Browser: Edge / Brave / Chrome / Firefox
        
If I open a modal window in which a link to another page exists, follow the link and then go back via browser history back, the modal window remains open and cannot be closed.

I have another link which opens a new modal. That link works as normal. So I think JavaScript still works, but not the Bootstrap JavaScript I think.

It's quite possible that it's because of Ruby on Rails and not Bootstrap. It may also have to do with the bfcache. I have no idea how to solve the problem or track down the error.

```
<button type="button" class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#exampleModal">
      Launch demo modal
    </button>
    
    <div class="modal fade" id="exampleModal" data-bs-backdrop="static" tabindex="-1" aria-labelledby="exampleModalLabel" aria-hidden="true">
      <div class="modal-dialog">
        <div class="modal-content">
          <div class="modal-header">
            <h5 class="modal-title" id="exampleModalLabel">Modal title</h5>
            <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
          </div>
          <div class="modal-body">
            <a href="/">Next Link to...</a>
            <a data-remote="true" href="/de/people/111">Test JavaScript still works</a>
          </div>
    
          <div class="modal-footer">
            <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Close</button>
            <button type="button" class="btn btn-primary">Save changes</button>
          </div>
        </div>
      </div>
    </div>
```

**Comment by nscherzer** [comment]
Same problem with Bootstrap 4.0 and Rails 5.2.6. Did you find a solution @chriskrams ?
